tests/unit/httpd_test.cc

```diff
@@ -205,6 +205,24 @@ SEASTAR_TEST_CASE(test_decode_url) {
     return make_ready_future<>();
 }
 
+SEASTAR_TEST_CASE(test_decode_path) {
```

Comment:
optional: Looks like the old behavior isn't tested by `test_decode_url`. Might be a good idea to add a check there as well.